File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.java
class Solution {
    public char repeatedCharacter(String s) {
        int[] letterInventory = new int[26];
        for (int i = 0; i < s.length(); i++) {
            char current = s.charAt(i);
            if (letterInventory[current - 97] > 0) {
                return current;
            }
            
            letterInventory[current - 97]++;
        }
        
        return ' ';
    }
}

File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
        int letterInventory[26] = {0};
        for (int i = 0; i < s.length(); i++) {
            if (letterInventory[s[i] - 97]) {
                return s[i];
            }
            
            letterInventory[s[i] - 97]++;
        }
        
        return ' ';
    }
};